otter_welcome_buddy/cogs/new_user_joins.py

```diff
@@ -3,31 +3,38 @@
 """
 from discord.ext import commands
 
+from otter_welcome_buddy.formatters import debug, messages
 
+
+# We need to create easier test by migrating everything to protected methods
 class Greetings(commands.Cog):
     """Custom message"""
 
-    def __init__(self, bot):
+    def __init__(self, bot, messages_dependency, debug_dependency):
         self.bot = bot
+        self.messages_formatter = messages_dependency
+        self.debug_formatter = debug_dependency
+
+    def _command_message(self):
+        return self.messages_formatter.welcome_message()
```

Comment:
Why did u change from sending a hello answer to resend the welcome message?

Reply:
Can you elaborate?

Reply:
Yeah, before you sent:
`await ctx.send(f"Hello {member.name}... This feels familiar.")`
Now you just send the welcome message injected.